src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.CopyFile.OSX.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO
+{
+    internal static partial class FileSystem
+    {
+        public static partial void CopyFile(string sourceFullPath, string destFullPath, bool overwrite)
+        {
+            // Fail fast for blatantly copying onto self
+            if (sourceFullPath == destFullPath)
+            {
+                throw new IOException(SR.Format(overwrite ? SR.IO_SharingViolation_File : SR.IO_FileExists_Name, destFullPath));
```

Comment:
This throws `IO_FileExists_Name` but we don't know the file exists.

Reply:
I fixed this, but don't know if I need another separate error for trying to copy a directory. If this was also needed, then it feels like it should probably also be removed at that point.

Reply:
Actually, I'll probably just delete it now as a part of implementing the next suggestion.